gempy/assets/geographic.py

```diff
@@ -0,0 +1,562 @@
+
+        # check that coordinates are in latlong, if not: convert
+        if self.type == 'utm':
+            self.utm_to_latlong()
+
+        for ps in self.point_sets:
+            fail = True
+            for geotiff in self.geotiffs:
+                try:
+                    ps.get_z_values_from_geotiff(geotiff)
+                except IndexError:
+                    continue
+                fail = False
+
+            # if point can not be detected: remove (default) or raise error
+            # if self.auto_remove = False
+
+            if fail:
+                if self.auto_remove:
+                    if self.debug:
+                        print("Point outside geotiff, drop")
+                    self.point_sets.remove(ps)
+                else:
+                    raise IndexError("Point outside of defined geotiffs!\nPlease define\
+                                     suitable geotiff or remove point (set self.auto_remove = True)")
+
+    def fit_plane_to_all_sets(self):
+        """Fit plane to all point sets
+
+        Results are stored in point set object (self.ctr, self.normal)
+        """
+        if self.type == 'latlong':
+            self.latlong_to_utm()
+
+        for ps in self.point_sets:
+            ps.plane_fit()
+            ps.get_orientation()
+
+    def stereonet(self):
+        """Create stereonet plot of all plane pole and half circle for all planes"""
+        import matplotlib.pyplot as plt
+        import mplstereonet
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection='stereonet')
+
+        # ax.plane(dip_dirs, dips, 'g-', linewidth=0.5)
+        for ps in self.point_sets:
+            ax.pole(ps.dip_direction - 90, ps.dip, 'gs', markersize=4)
+            ax.plane(ps.dip_direction - 90, ps.dip, 'g', markersize=4)
+
+        # ax.rake(strike, dip, -25)
+        ax.grid()
+
+    def add_geotiff(self, geotiff):
+        """Add geotiff to list of geotiffs (self.geotiffs)
+
+        Args:
+            geotiff (filename) : filename (with complete path) to geotiff
+        """
+        if self.debug:
+            print("Note: for efficiency reasons, add the most important geotiff first!")
+        self.geotiffs.append(geotiff)
+
+    def latlong_to_utm(self):
+        """Convert all points from lat long to utm"""
+        if self.type == 'latlong':  # else not required...
+            if self.debug:
+                print("Convert Lat/Long to UTM")
+            for ps in self.point_sets:
+                ps.latlong_to_utm()
+            self.type = 'utm'
+
+    def utm_to_latlong(self):
+        """Convert all points from utm to lat long"""
+        if self.type == 'utm':  # else not required...
+            if self.debug:
+                print("Convert UTM to Lat/Long")
+            for ps in self.point_sets:
+                ps.utm_to_latlong()
+            self.type = 'latlong'
+
+
+def get_pixel_values(geo_coords, data_source, **kwds):
```

Comment:
rasterio can reduce this to a one-liner

Reply:
Yes, I have actually seen that before and wanted to implement it :-) Seems like the right time, now...